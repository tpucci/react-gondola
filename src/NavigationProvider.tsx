import React from 'react';
import { Provider } from 'mobx-react/native';

interface IProps {
  children: React.ReactNode;
}
export const NavigationProvider = ({ children }: IProps) => (
  <Provider navigation={{}}>{React.Children.only(children)}</Provider>
);
