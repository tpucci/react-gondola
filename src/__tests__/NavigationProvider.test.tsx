import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'mobx-react/native';
import { View } from 'react-native';

import { NavigationProvider } from '../NavigationProvider';

describe('NavigationProvider', () => {
  it('renders a MobX provider with a navigation prop', () => {
    const testRenderer = TestRenderer.create(
      <NavigationProvider>
        <View />
      </NavigationProvider>
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
    const testInstance = testRenderer.root;
    const providerInstance = testInstance.findByType(Provider);
    expect(providerInstance.props.navigation).toBeDefined();
  });
});
