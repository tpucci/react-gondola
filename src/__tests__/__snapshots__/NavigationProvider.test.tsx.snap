// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationProvider renders a MobX provider with a navigation prop 1`] = `<View />`;
